### PR TITLE
doc: Generate v1.15.0

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

This PR is to generate changelog for v1.15.0.

Link to any related issue(s): CLOUDP-227250

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Comment by lantoli** [comment]
in PR description you can say just CLOUDP-227250 instead of the full url/link

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,147 @@
...
+- Error: json: cannot unmarshal number 9501614080 into Go struct field CloudProviderSnapshot.storageSizeBytes of type int  [\#1341](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1341)
+- [Query] Unable to find the `endpoint_service_id` [\#1281](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1281)
+
+**Internal Improvements**
+
+- chore: Changes from env variable to input in Import GPG key [\#1898](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1898) ([oarbusi](https://github.com/oarbusi))
+- chore: Upgrades `mongodbatlas_custom_db_role` resource to auto-generated SDK [\#1896](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1896) ([lantoli](https://github.com/lantoli))
+- chore: Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.31.0 to 2.32.0 [\#1893](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1893) ([dependabot[bot]](https://github.com/apps/dependabot))
+- chore: Bump github.com/aws/aws-sdk-go from 1.50.3 to 1.50.7 [\#1892](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1892) ([dependabot[bot]](https://github.com/apps/dependabot))
```
Comment (lantoli) [inline CHANGELOG.md]: recommend bumps together at the end of the section

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,147 @@
 # Changelog
 
+## [v1.15.0-pre](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.15.0-pre) (2024-02-01)
+
+[Full Changelog](https://github.com/mongodb/terraform-provider-mongodbatlas/compare/v1.14.0...v1.15.0-pre)
+
+**Breaking changes:**
+
+- remove!: Removes `page_num` and `items_per_page` attributes in `mongodbatlas_search_indexes` data source [\#1880](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1880) ([AgustinBettati](https://github.com/AgustinBettati))
+- remove!: Removal Cloud Provider Access Resource and Data source [\#1804](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1804) ([andreaangiolillo](https://github.com/andreaangiolillo))
```
Comment (lantoli) [inline CHANGELOG.md]:
```suggestion
- remove!: Removes `cloud_provider_access` resource and data source [\#1804](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1804) ([andreaangiolillo](https://github.com/andreaangiolillo))
```

File: CHANGELOG.md (hunk 1 above)
Comment (oarbusi) [reply CHANGELOG.md]: Thanks for the suggestion. Done

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,147 @@
...
+- fix: Improves error message when improperly setting `provider_region_name` field [\#1815](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1815) ([maastha](https://github.com/maastha))
+- fix: Provider produces inconsistent result after importing `mongodbatlas_encryption_at_rest` [\#1813](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1813) ([andreaangiolillo](https://github.com/andreaangiolillo))
+- fix: Update attributes `copy_protection_enabled`, `pit_enabled` and `encryption_at_rest_enabled` in the  resource `mongodbatlas_backup_compliance_policy` to be Optional [\#1803](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1803) ([andreaangiolillo](https://github.com/andreaangiolillo))
+- fix: Incompatible schema defined for `mongodbatlas_backup_compliance_policy` [\#1799](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1799) ([andreaangiolillo](https://github.com/andreaangiolillo))
+- fix: Fix `mongodbatlas_clusters` plural data source to set `auto_scaling_disk_gb_enabled` attribute correctly [\#1722](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1722) ([maastha](https://github.com/maastha))
```
Comment (lantoli) [inline CHANGELOG.md]:
```suggestion
- fix: Fixes `mongodbatlas_clusters` plural data source to set `auto_scaling_disk_gb_enabled` attribute correctly [\#1722](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1722) ([maastha](https://github.com/maastha))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,147 @@
...
+**Enhancements**
+
+- feat: Adds support to new Federated Auth parameters for OIDC  [\#1874](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1874) ([oarbusi](https://github.com/oarbusi))
+- feat: Adds new `ip_addresses` computed attribute in `mongodbatlas_project` resource and data sources [\#1850](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1850) ([AgustinBettati](https://github.com/AgustinBettati))
+- feat: Adds atlas Organization New Parameters Support  [\#1835](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1835) ([maastha](https://github.com/maastha))
+
+**Bug Fixes**
+
+- `mongodbatlas_project_ip_access_list` Unexpected replacement of CIDR with IP address [\#1571](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1571)
+- fix: Add toUpperCase to provider and region fields in cluster resources [\#1837](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1837) ([andreaangiolillo](https://github.com/andreaangiolillo))
```
Comment (lantoli) [inline CHANGELOG.md]:
```suggestion
- fix: Adds toUpperCase to provider and region fields in cluster resources [\#1837](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1837) ([andreaangiolillo](https://github.com/andreaangiolillo))
```

**Review by lantoli — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,149 @@
...
+**Breaking changes:**
+
+- remove!: Removes `page_num` and `items_per_page` attributes in `mongodbatlas_search_indexes` data source [\#1880](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1880) ([AgustinBettati](https://github.com/AgustinBettati))
+- remove!: Removes `cloud_provider_access`  resource and data source [\#1804](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1804) ([andreaangiolillo](https://github.com/andreaangiolillo))
+
+**Enhancements**
+
+- feat: Adds support to new Federated Auth parameters for OIDC  [\#1874](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1874) ([oarbusi](https://github.com/oarbusi))
+- feat: Adds new `ip_addresses` computed attribute in `mongodbatlas_project` resource and data sources [\#1850](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1850) ([AgustinBettati](https://github.com/AgustinBettati))
+- feat: Adds atlas Organization New Parameters Support  [\#1835](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1835) ([maastha](https://github.com/maastha))
```
Comment (Zuhairahmed) [inline CHANGELOG.md]:
```suggestion
- feat: Adds `mongodbatlas_organization` New Parameters Support  [\#1835](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1835) ([maastha](https://github.com/maastha))
```

**Review by Zuhairahmed — COMMENTED** [review]
@oarbusi overall looks good, but 2 quick questions: 
1. Can you update v1.15.0 upgrade guide for Breaking Changes and other Enhancements? 
2. Did we check as part of this release to ensure all parameters / resources flagged for code removal in v1.15.0 were indeed removed as part of this release?

**Comment by oarbusi** [comment]
> @oarbusi overall looks good, but 2 quick questions:

> 1. Can you update v1.15.0 upgrade guide for Breaking Changes and other Enhancements?

@Zuhairahmed Both breaking changes in this release are removals, one was already in the `Deprecations and Removals:` section. I'll add the other one there. Or should I put both in `breaking changes` section of upgrade guide? 

> 2. Did we check as part of this release to ensure all parameters / resources flagged for code removal in v1.15.0 were indeed removed as part of this release?

No mentions of anything to be deprecated in 1.15.0

**Review by pierwill — APPROVED** [review]

**Review by marcosuma — APPROVED** [review]

**Comment by Zuhairahmed** [comment]
> > @oarbusi overall looks good, but 2 quick questions:
> 
> > 1. Can you update v1.15.0 upgrade guide for Breaking Changes and other Enhancements?
> 
> @Zuhairahmed Both breaking changes in this release are removals, one was already in the `Deprecations and Removals:` section. I'll add the other one there. Or should I put both in `breaking changes` section of upgrade guide?
> 
> > 2. Did we check as part of this release to ensure all parameters / resources flagged for code removal in v1.15.0 were indeed removed as part of this release?
> 
> No mentions of anything to be deprecated in 1.15.0

Thanks, just looks odd with nothing after "Breaking Changes:" if nothing you want to call out here suggest removing this line 

image

**Review by Zuhairahmed — APPROVED** [review]

**Pull request merged**